dev/com.ibm.ws.security.oauth/test/com/ibm/ws/security/oauth20/plugins/db/CachedDBOidcClientProviderXORTest.java

```diff
@@ -0,0 +1,62 @@
+/*******************************************************************************
+ * Copyright (c) 2014, 2019 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package com.ibm.ws.security.oauth20.plugins.db;
+
+import javax.naming.Context;
+
+import org.junit.Before;
+import org.junit.BeforeClass;
+
+import test.common.SharedOutputManager;
+
+/**
+ * This unit test is running with XOR enabled for the client secret
+ */
+public class CachedDBOidcClientProviderXORTest extends CachedDBOidcClientProviderTest {
+
+    @BeforeClass
+    public static void setUpBeforeClass() throws Exception {
+        setHash(false);
+
+        outputMgr = SharedOutputManager.getInstance();
+        outputMgr.captureStreams();
+
+        System.setProperty(Context.INITIAL_CONTEXT_FACTORY, InitialContextFactoryMock.class.getName());
+
+        SAMPLE_CLIENTS = getsampleOidcBaseClients(5, PROVIDER_NAME);
+    }
+
+    @Override
+    @Before
+    public void setupBefore() {
```

Comment:
It looks like this method could actually stay in the abstract `CachedDBOidcClientProviderTest` as-is; the appropriate overridden `invokeConstructorAndInitialize()` method would still be invoked at runtime, so everything should execute correctly.